researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider2.java

```diff
@@ -0,0 +1,285 @@
+
+import org.researchstack.backbone.DataProvider;
+import org.researchstack.backbone.DataResponse;
+import org.researchstack.backbone.model.ConsentSignatureBody;
+import org.researchstack.backbone.model.SchedulesAndTasksModel;
+import org.researchstack.backbone.model.User;
+import org.researchstack.backbone.result.TaskResult;
+import org.researchstack.backbone.task.Task;
+import org.sagebionetworks.bridge.android.manager.BridgeManagerProvider;
+import org.sagebionetworks.bridge.android.manager.auth.AuthenticationManager;
+import org.sagebionetworks.bridge.rest.model.SharingScope;
+import org.sagebionetworks.bridge.rest.model.StudyParticipant;
+import org.sagebionetworks.bridge.rest.model.UserSessionInfo;
+
+import rx.Completable;
+import rx.Observable;
+import rx.Observer;
+import rx.functions.Action0;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sagebionetworks.bridge.researchstack.ApiUtils.SUCCESS_DATA_RESPONSE;
+
+/**
+ * Created by jyliu on 1/20/2017.
+ */
+@AnyThread
+public class BridgeDataProvider2 extends DataProvider {
+    private final BridgeManagerProvider bridgeManagerProvider;
+
+    public BridgeDataProvider2(BridgeManagerProvider bridgeManagerProvider) {
+        this.bridgeManagerProvider = bridgeManagerProvider;
+    }
+
+    @Override
+    @Deprecated
+    public Observable<DataResponse> initialize(@NonNull Context context) {
+        checkNotNull(context);
+
+        return Observable.just(new DataResponse(true, "donothing"));
+    }
+
+    @Override
+    @NonNull
+    @Deprecated
+    public Observable<DataResponse> signUp(@NonNull Context context, @NonNull String email,
+                                           @NonNull String username, @NonNull String password) {
+        return signUp(email, password);
+    }
+
+    @NonNull
+    public Observable<DataResponse> signUp(@NonNull String email, @NonNull String password) {
+        checkNotNull(email);
+        checkNotNull(password);
+
+        return bridgeManagerProvider.getAuthenticationManager()
+                .signUp(email, password)
+                .andThen(SUCCESS_DATA_RESPONSE);
+    }
+
+    @Override
+    @Deprecated
+    public boolean isSignedUp(Context context) {
+        return isSignedUp();
+    }
+
+    public boolean isSignedUp() {
+        return bridgeManagerProvider.getAuthenticationManager()
+                .getDao()
+                .getEmail() != null;
+    }
+
+    @Override
+    @Nullable
+    @Deprecated
+    public User getUser(Context context) {
+        return getUser();
+    }
+
+    @Override
+    public void setUser(Context context, User user) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Nullable
+    public User getUser() {
+        AuthenticationManager.DAO DAO =
+                bridgeManagerProvider.getAuthenticationManager()
+                        .getDao();
+
+        User user = new User();
+        user.setEmail(DAO.getEmail());
```

Comment:
I think there should be one access point for a User object.  Where that is, doesn't really matter, but it will create confusion otherwise when this class only returns a User with their email set, instead of all the user fields.